src/utilities/playlistsUtils.js

```diff
@@ -1,4 +1,120 @@
 import axios from "axios";
 import { playlist } from "./apiUrls";
 
-playlist;
+const setPlaylist = async () => {
+  try {
+    const { data } = await axios.get(playlist, {
+      headers: {
+        authorization: localStorage.getItem("token"),
+      },
+    });
+    return data.playlists;
+  } catch (error) {
+    console.error(error);
+  }
+};
+
+const addSinglePlaylist = async (singlPlaylist, video, playlistDispatch) => {
+  try {
+    const { data } = await axios.post(
+      playlist,
+      {
+        playlist: singlPlaylist,
+      },
+      {
+        headers: {
+          authorization: localStorage.getItem("token"),
+        },
+      }
+    );
+    playlistDispatch({ type: "ADD_SINGLE_PLAYLIST", payload: data.playlists });
+    await postSingleVideoInAPlaylist(
+      data.playlists[data.playlists.length - 1]._id,
+      video,
+      playlistDispatch
+    );
+  } catch (error) {
+    console.error(error);
+  }
+};
+
+const deleteSinglePlaylist = async (playlistId, playlistDispatch) => {
+  try {
+    const { data } = await axios.delete(`${playlist}/${playlistId}`, {
+      headers: {
+        authorization: localStorage.getItem("token"),
+      },
+    });
+    playlistDispatch({
+      type: "DELETE_SINGLE_PLAYLIST",
+      payload: data.playlists,
+    });
+  } catch (error) {
+    console.error(error);
+  }
+};
+
+const getSinglePlaylistData = async (playlistId) => {
+  try {
+    const { data } = await axios.get(`${playlist}/${playlistId}`, {
+      headers: {
+        authorization: localStorage.getItem("token"),
+      },
+    });
+
+    return data.playlist;
+  } catch (error) {
+    console.error(error);
+  }
+};
+
+const postSingleVideoInAPlaylist = async (
+  playlistId,
+  video,
+  playlistDispatch
+) => {
+  try {
+    const { data } = await axios.post(
+      `${playlist}/${playlistId}`,
+      { video },
+      {
+        headers: {
+          authorization: localStorage.getItem("token"),
+        },
+      }
+    );
+    playlistDispatch({
+      type: "ADD_SINGLE_VIDEO_TO_PLAYLIST",
+      payload: data.playlist,
+    });
+  } catch (error) {
+    console.error(error);
+  }
+};
+
+const deleteVideoInPlaylist = async (playlistId, videoId, playlistDispatch) => {
+  try {
+    const { data } = await axios.delete(
+      `${playlist}/${playlistId}/${videoId}`,
+      {
+        headers: {
+          authorization: localStorage.getItem("token"),
```

Comment:
you can make a function for token and use it everywhere

Reply:
But I will have to import it everywhere and also write that in front of authorization. Thats why I wrote like this only